flake8diff/vcs/hg.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import unicode_literals, print_function
+
+import logging
+import subprocess
+
+from ..utils import _execute
+from .base import VCSBase
+
+
+logger = logging.getLogger(__name__)
+
+
+class HgVCS(VCSBase):
+    """
+    Mercurial support implementation
+    """
+    name = 'hg'
+
+    def get_vcs(self):
+        """
+        Get git binary executable path
+        """
+        vcs = _execute('which hg', strict=True).strip()
+        self._check_extdiff_extension(vcs)
+        return vcs
+
+    def is_used(self):
+        """
+        Determines if this VCS should be used
+
+        TODO: implement
+        """
+        return True
```

Comment:
Probably a good idea to implement this by returning boolean if the current folder is part of mercurial repository. Also then its probably a good idea to change git as well.

By implementing `is_used`, that should enable auto-detection which will not require passing `--vcs` parameter.

Sample implementation (not tested):

```python
try:
    _execute('hg status', strict=True)
except subprocess.CalledProcessError:
    return False
else:
    return True
```

same should work for git